### PR TITLE
feat: Add MCP integration support for Slack and Twitter

### LINKED ISSUE #36: MCP support to RAG on other software
People want to RAG on their twitter bookmark

Feedback from the user

### PR DESCRIPTION
**Pull request by ASuresh0524**
- Implement SlackMCPReader for connecting to Slack MCP servers
- Implement TwitterMCPReader for connecting to Twitter MCP servers
- Add SlackRAG and TwitterRAG applications with full CLI support
- Support live data fetching via Model Context Protocol (MCP)
- Add comprehensive documentation and usage examples
- Include connection testing capabilities with --test-connection flag
- Add standalone tests for core functionality
- Update README with detailed MCP integration guide
- Add Aakash Suresh to Active Contributors

Resolves #36 

## What does this PR do?

This PR adds comprehensive MCP (Model Context Protocol) integration support to LEANN, enabling users to perform RAG on live data from various platforms through standardized MCP servers. The implementation includes robust readers for Slack messages and Twitter bookmarks, complete RAG applications with CLI support, and extensive documentation.

**Key Features:**

**Slack MCP Integration:**
- **SlackMCPReader**: Connects to Slack MCP servers to fetch live message data with proper authentication handling
- **SlackRAG**: Full RAG application with configurable workspace filtering, channel selection, and conversation grouping
- Connection testing with `--test-connection` flag for easy debugging

**Twitter MCP Integration:**
- **TwitterMCPReader**: Connects to Twitter MCP servers to fetch bookmark data with engagement metadata support
- **TwitterRAG**: Complete RAG application with bookmark filtering, content processing options, and metadata inclusion
- Flexible processing modes for content-only or metadata-rich indexing

**Comprehensive Documentation:**
- Updated README with detailed MCP integration guide including setup instructions and troubleshooting
- Usage examples for both Slack and Twitter integrations
- Expandable sections with example queries and platform extension guidelines

**Testing & Examples:**
- `examples/mcp_integration_demo.py`: Interactive demonstration script showing MCP capabilities
- `tests/test_mcp_standalone.py`: Comprehensive standalone tests verifying core functionality without dependencies
- All tests pass successfully with proper error handling and edge case coverage

**Live Data Benefits:**
- Real-time data access without manual exports
- Standardized protocol works with any MCP-compatible server
- Easy extension framework for adding new platforms
- Secure authentication handled by MCP servers

**Usage Examples:**

```bash
# Test Slack MCP connection
python -m apps.slack_rag --mcp-server "slack-mcp-server" --test-connection

# Index Slack messages
python -m apps.slack_rag \
  --mcp-server "slack-mcp-server" \
  --workspace-name "my-team" \
  --channels general dev-team \
  --query "What did we decide about the product launch?"

# Index Twitter bookmarks
python -m apps.twitter_rag \
  --mcp-server "twitter-mcp-server" \
  --max-bookmarks 1000 \
  --query "What AI articles did I bookmark?"
```

## Related Issues

Fixes #36

## Checklist

- [x] Tests pass (`uv run pytest`) - Created and ran comprehensive standalone tests for all MCP functionality
- [x] Code formatted (`ruff format` and `ruff check`) - All linting issues resolved, follows PEP 8 conventions
- [x] Pre-commit hooks pass (`pre-commit run --all-files`) - Code quality checks completed for all new modules

### CODE REVIEW

**Comment by yichuan-w** [comment]
Hi @ASuresh0524, good job. Have you tested these features yourself

**Comment by ASuresh0524** [comment]
Yes, I have! Tested it locally, it should work. Fixing some errors in the pr though as of right now.

File: README.md [hunk 1]
```diff
@@ -56,24 +56,282 @@ LEANN achieves this through *graph-based selective recomputation* with *high-deg
 curl -LsSf https://astral.sh/uv/install.sh | sh
 ```
 
-### 🚀 Quick Install
-
-Clone the repository to access all examples and try amazing applications,
+### 🚀 Quick Install (Recommended for Most Users)
 
+**Step 1: Clone and Setup**
```
Comment (yichuan-w) [inline README.md]: Remember, we do not need all of these, keep the readme as simple as possible and I think we only need the line 1305 stuff. and I guess we do not need the stuff from 61 to 334 which you add, also these seem to be generated by curosr

Comment (ASuresh0524) [reply README.md]: Sorry, sounds good, will keep it as simple as possible. Yeah, first time using cursor wasn't too sure how accurate it would be for the documentation, think I'll just make these edits manually then. Sorry about that.

**Comment by ASuresh0524** [comment]
should be updated. please let me know if it looks okay now

File: README.md [hunk 2]
```diff
@@ -48,15 +48,15 @@ LEANN achieves this through *graph-based selective recomputation* with *high-deg
 
 ## Installation
 
-### 📦 Prerequisites: Install uv
+### Prerequisites: Install uv
```
Comment (yichuan-w) [inline README.md]: Why change here?

File: README.md [hunk 3]
```diff
@@ -48,15 +48,15 @@ LEANN achieves this through *graph-based selective recomputation* with *high-deg
...
 
 [Install uv](https://docs.astral.sh/uv/getting-started/installation/#installation-methods) first if you don't have it. Typically, you can install it with:
 
 ```bash
 curl -LsSf https://astral.sh/uv/install.sh | sh
 ```
 
-### 🚀 Quick Install
+### Quick Install
```
Comment (yichuan-w) [inline README.md]: same

File: README.md (hunk 2 above)
Comment (yichuan-w) [reply README.md]: Also, add the feature of MCP in Line 23 and link to the section, briefly introduce slack and twitter stuff

File: README.md [hunk 4]
```diff
@@ -774,6 +775,147 @@ Once your iMessage conversations are indexed, you can search with queries like:
...
+  --mcp-server "twitter-mcp-server" \
+  --max-bookmarks 1000 \
+  --query "What AI articles did I bookmark about machine learning?"
+```
+
+**Setup Requirements:**
+1. Install a Twitter MCP server (e.g., `npm install -g twitter-mcp-server`)
+2. Configure Twitter API credentials:
+   ```bash
+   export TWITTER_API_KEY="your-api-key"
```
Comment (yichuan-w) [inline README.md:851]: need to tell users more about how to get these?

File: README.md [hunk 5]
```diff
@@ -774,6 +775,147 @@ Once your iMessage conversations are indexed, you can search with queries like:
...
+  --mcp-server "slack-mcp-server" \
+  --workspace-name "my-team" \
+  --channels general dev-team random \
+  --query "What did we decide about the product launch?"
+```
+
+**Setup Requirements:**
+1. Install a Slack MCP server (e.g., `npm install -g slack-mcp-server`)
+2. Configure Slack API credentials:
+   ```bash
```
Comment (yichuan-w) [inline README.md:812]: need to tell users more about how to get these?

File: README.md (hunk 2 above)
Comment (ASuresh0524) [reply README.md]: Sorry, I thought I put the emoji after the change lol, thought it looked cursor'ed. same for the other one, can change to keep those two will put it back and change as needed.

**Comment by yichuan-w** [comment]
LGTM
